components/smalltalk/squeak5/Makefile

```diff
@@ -0,0 +1,95 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"). You may
+# only use this file in accordance with the terms of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source. A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Package Makefile for the "OpenSmalltalk Cog Spur" Squeak Smalltalk V5 system
+# See http://squeak.org and http://opensmalltalk.org
+#
+
+BUILD_BITS=64
+COMPILER=gcc
+
+include ../../../make-rules/shared-macros.mk
+
+COMPONENT_NAME=		squeak5
+COMPONENT_VERSION=	5.0.6
+COMPONENT_SUMMARY=	The Squeak V5 Smalltalk System
+COMPONENT_PROJECT_URL=	http://www.squeak.org
+COMPONENT_FMRI=		runtime/squeak5
+COMPONENT_CLASSIFICATION=	Development/Other Languages
+COMPONENT_LICENSE=	LICENSE
+COMPONENT_SRC=		opensmalltalk-vm
+
+GIT=			git
+GIT_REPO=		https://github.com/cstes/opensmalltalk-vm.git
+GIT_BRANCH=		sun
+GIT_TAG=		sun-v5.0.6
+
+TEST_TARGET= $(NO_TESTS)
+
+include $(WS_MAKE_RULES)/common.mk
+
+PATH=$(PATH.gnu)
+
+# The ugly hack with update-publish target is necessary to update
+# source from git repository on each "gmake publish".
+publish: update-publish
+
```

Comment:
Why is it necessary to update the sources from git on each "gmake publish"? This would create a new build on every Jenkins run on our build server, thus, whenever a change in our git repository happens.

Reply:
Ideally I could build from an 'archive' or 'release' from codeload.github.com  (a tar.gz package).  I tested that and it is better.  But unfortunately I think the opensmalltalk build setup currently more or less implies that you have to build from git clones (not unversion-ed sources).  So that's why I  "git clone" the sources.   The "update-publish" target is not doing anything for the moment in my case.   I copied and modified that code from "./openindiana/time-slider/Makefile" ; I'll check this later ...

Reply:
I think alp should have a look at it. I am not experienced enough in this area.